pallets/dao-assets/src/functions.rs

```diff
@@ -68,7 +114,18 @@ impl<T: Config> Pallet<T> {
 		who: impl Borrow<T::AccountId>,
 		block: BlockNumberFor<T>,
 	) -> Option<T::Balance> {
-		Self::search_history(AccountHistory::<T>::get(id, who.borrow()), block)
+		// Self::search_history(AccountHistory::<T>::get(id, who.borrow()), block)
+		let mut nearest_block = block;
+		let mut final_balance = Some(0_u32.into());
+
+		for (block_num, balance) in AccountHistory::<T>::iter_prefix((id, who.borrow())) {
+			if block_num >= nearest_block {
```

Comment:
i'm not sure what you are doing here. Why `>=`, wouldn't we want to get the closest from the other side?